psycop/common/feature_generation/data_checks/flattened/feature_describer_tsflattener_v2.py

```diff
@@ -0,0 +1,98 @@
+import re
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import polars as pl
+
+
+@dataclass
+class ParsedPredictorColumn:
+    col_name: str
+    feature_name: str
+    fallback: str
+    time_interval_start: str
+    time_interval_end: str
+    time_interval_format: str
+    resolve_multiple_strategy: str
+    is_static: bool
+
+
+def tsflattener_v2_column_is_static(col_name: str) -> bool:
+    str_match = "within"
+    return str_match not in col_name
+
+
+def get_value_proportion(series: pl.Series, value: str) -> float:
+    """Get proportion of series that is equal to the value argument."""
+    if value == "nan":
+        return round(series.is_nan().mean(), 2)  # type: ignore
+    return round(series.eq(float(value)).mean(), 2)  # type: ignore
+
+
+def parse_predictor_column_name(
+    col_name: str,
+    is_static: Callable[[str], bool] = tsflattener_v2_column_is_static,
+    feature_name_regex: str = r"[a-zA-Z]+_(.*?)_(?=fallback|within)",
+    fallback_regex: str = r"_fallback_(.+)$",
+    time_interval_start_regex: str = r"_within_([0-9]+)",
+    time_interval_end_regex: str = "_to_([0-9]+)_",
+    time_interval_format_regex: str = r"to_[0-9]+_([a-z]+)_",
+    resolve_multiple_strategy_regex: str = r"([a-z]+)_fallback",
+) -> ParsedPredictorColumn:
+    col_is_static = is_static(col_name)
+    feature_name = re.search(feature_name_regex, col_name).group(1)  # type: ignore
+    fallback = re.search(fallback_regex, col_name).group(1)  # type: ignore
+    time_interval_start = (
+        re.search(time_interval_start_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    time_interval_end = (
+        re.search(time_interval_end_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    time_interval_format = (
+        re.search(time_interval_format_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    resolve_multiple_strategy = (
+        re.search(resolve_multiple_strategy_regex, col_name).group(1)  # type: ignore
+        if not col_is_static
+        else "N/A"
+    )
+    return ParsedPredictorColumn(
+        col_name=col_name,
+        feature_name=feature_name,
+        fallback=fallback,
+        time_interval_start=time_interval_start,
+        time_interval_end=time_interval_end,
+        time_interval_format=time_interval_format,
+        resolve_multiple_strategy=resolve_multiple_strategy,
+        is_static=is_static(col_name),
+    )
+
+
+def generate_feature_description_df(
+    df: pl.DataFrame,
+    column_name_parser: Callable[[str], ParsedPredictorColumn] = parse_predictor_column_name,
+) -> pl.DataFrame:
+    feature_rows = []
+    for col in df.columns:
+        parsed_col = column_name_parser(col)
+        n_unique = df[col].n_unique()
+        proportion_missing = round(df[col].is_nan().mean(), 2)  # type: ignore
```

Comment:
so this is only if we chose nan as fallback, so functionally the same as "proportion using fallback" for those cols? and otherwise always 0%? seems a bit redundant in that case - maybe more interesting how many were missing before fallback imputation? but i guess that's what the "proportion using fallback" does?

Reply:
Yeah, let's remove proportion missing, it's redundant